### ISSUE TITLE
Rule Change: (`array-callback-return`) support `Array.prototype.findLast` and `Array.prototype.findLastIndex`

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#16313]
### What rule do you want to change?

`array-callback-return`

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
foo.findLast(function() {});
foo.findLastIndex(function() {});
```

### What does the rule currently do for this code?

Ignores Stage 4 `Array.prototype.findLast` and `Array.prototype.findLastIndex`

### What will the rule do after it's changed?

Enforce return statements in callbacks of `Array.prototype.findLast` and `Array.prototype.findLastIndex`

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

- https://github.com/tc39/proposal-array-find-from-last

**Comment by mdjermanovic** [comment]
Thanks for the issue! Marked as accepted.

Per our [semver policy](https://github.com/eslint/eslint#semantic-versioning-policy), this can be included in a minor release.

> A new addition to an existing rule to support a newly-added language feature (within the last 12 months) that will result in ESLint reporting more linting errors by default.
